### ISSUE TITLE
[Bug] PD kv-consumer MTP placeholder draft token can crash rejection sampling

### ISSUE THREAD

**Issue by QwertyJack** [vllm-project/vllm-ascend#10045]
## Summary

In PD disaggregated serving, the decode-side kv-consumer MTP first step can schedule `PLACEHOLDER_TOKEN_ID` (`-1`) as a placeholder draft token. That placeholder is required for the D-side MTP/full-graph shape, but it is not a real draft token ID.

The Ascend random rejection sampler currently can treat that placeholder as a vocabulary/probability-table index. On Ascend this can crash the D worker with a vector core exception when the sampler kernel reads probability tables using token id `-1`.

Related fix: #10043

## Affected path

Observed with a DeepSeek V4 Flash PD setup where the D node is a kv-consumer and MTP/speculative decode is enabled through the recompute scheduler path.

The code-level trigger is:

- `RecomputeScheduler.is_mtp_kv_consumer == True`
- `num_spec_tokens > 0`
- a new PD kv-consumer request on its first D step
- scheduler output contains placeholder draft tokens such as `scheduled_spec_decode_tokens={req_id: [-1]}`
- the request uses the random rejection sampler path (`draft_probs is None` in `NPUModelRunner._sample`)

## Symptom

The D node crashes during model execution. The first hardware fault in the local logs is:

```text
fault kernel_name=rejection_random_sample_kernel_1
ErrCode=507035, desc=[vector core exception]
```

The matching scheduler dump from the same DP worker shows requests scheduled with placeholder draft tokens:

```text
num_scheduled_tokens={...: 2, ...: 2, ...: 2}
scheduled_spec_decode_tokens={...: [-1], ...: [-1], ...: [-1]}
kv_connector_stats={'dp': 9}
```

The placeholder scheduling itself is expected for the D/MTP path; the bug is that rejection sampling consumes the placeholder as if it were a valid draft token id.

## Root cause

On the first D step of a PD kv-consumer request, `PLACEHOLDER_TOKEN_ID` is used to preserve the MTP/full-graph decode shape. The random rejection sampler should treat it as an invalid draft token and reject it directly.

Instead, the Ascend sampler can index probability tables with the placeholder:

1. `RecomputeScheduler.add_request()` schedules `[-1]` for the PD kv-consumer first step to preserve D/MTP shape.
2. Scheduler output contains `scheduled_spec_decode_tokens={req_id: [-1]}` and schedules two tokens for the request.
3. `NPUModelRunner._sample()` calls `AscendRejectionSampler` with `draft_probs=None`.
4. `rejection_random_sample_kernel` reads `draft_token_id=-1` and can use it in address calculations like `target_probs_ptr + token_idx * vocab_size + draft_token_id`.
5. The invalid index can trigger `rejection_random_sample_kernel_1` failure with `507035/vector core exception`.

The same semantic bug exists in Python fallback paths as negative indexing, where `-1` can be treated as the last vocabulary entry instead of a placeholder.

## Minimal reproducer

A code-level reproducer is covered by the regression tests in #10043:

```bash
pytest -q tests/ut/core/test_recompute_scheduler.py tests/ut/sample/test_rejection_sampler.py
```

The scheduler-side minimal setup confirms the required placeholder contract:

```python
scheduler.is_kv_producer = False
scheduler.is_hybrid_model = False
scheduler.is_mtp_kv_consumer = True
scheduler.num_spec_tokens = 1
scheduler.max_model_len = 1024

request = Request(
    request_id="pd-consumer-first-step",
    prompt_token_ids=[1, 2, 3, 4],
    sampling_params=SamplingParams(max_tokens=8),
    pooling_params=None,
)

scheduler.add_request(request)
assert request.spec_token_ids == [PLACEHOLDER_TOKEN_ID]
```

The sampler-side minimal setup is:

```python
output_token_ids = torch.full((1, 2), PLACEHOLDER_TOKEN_ID)
cu_num_draft_tokens = torch.tensor([1])
draft_token_ids = torch.tensor([PLACEHOLDER_TOKEN_ID])
target_probs = torch.tensor([[0.0, 0.0, 1.0]])
recovered_token_ids = torch.tensor([2])
uniform_probs = torch.tensor([0.0])
is_greedy = torch.tensor([False])

rejection_random_sample_pytorch(
    output_token_ids,
    cu_num_draft_tokens,
    draft_token_ids,
    None,
    target_probs,
    torch.tensor([[100]]),
    recovered_token_ids,
    uniform_probs,
    is_greedy,
    max_spec_len=1,
    vocab_size=3,
    IS_NGRAM=True,
)
```

Expected behavior: placeholder draft tokens are rejected directly and the recovered token is emitted; no code path should read probability tables with token id `-1`.

## Expected behavior

The first D step for a PD kv-consumer MTP request should keep placeholder draft tokens for graph shape, while rejection sampling must treat those placeholders as invalid draft tokens and avoid vocabulary/probability-table indexing with `-1`.
